scripts/lint_jjb.py

```diff
@@ -155,6 +164,113 @@ def check_retention(job, in_file):
     return return_value
 
 
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+def check_timed_trigger_value(trigger):
+
+    # We cannot parse macros, so we assume that
+    # they are ok.
+    if '{' in trigger and '}' in trigger:
+        return_value = 0
+    else:
+        # Jenkins supports the use of hashes, denoted by the symbol H,
+        # in the trigger to allow the load to be spread on the system.
+        # The crontab library does not know how to validate them, so
+        # we replace them with 0.
+        _trigger = trigger.replace('H', '0')
+
+        if crontab.CronSlices.is_valid(_trigger):
+            return_value = 0
+        else:
+            return_value = 1
+
+    return return_value
+
+
+def check_timed_trigger_list(data, in_file):
+    error_message = ("{f}/{n}: Valid timer trigger not found."
+                     " triggers: {t}\n"
+                     .format(n=data['name'], f=in_file,
+                             t=data.get('triggers')))
+
+    timed_trigger_list = jmespath.search('triggers[*].timed', data)
+
+    return_value = 0
+    for trigger in timed_trigger_list or []:
+        return_value = check_timed_trigger_value(trigger)
+        if return_value == 1:
+            sys.stderr.write(error_message)
+
+    return return_value
+
+
+def check_timed_trigger_cron(data, in_file):
+    error_message = ("{f}/{n}: Valid CRON value not found."
+                     " CRON: {t}\n"
+                     .format(n=data['name'], f=in_file,
+                             t=data.get('CRON')))
+
+    cron_value = data.get('CRON')
+
+    return_value = 0
+    if cron_value:
```

Comment:
If `cron_value` is `""` or `None` what would happen in the job?

Reply:
Both will fail with the changes I'll be adding shortly.